mergesort/mergesort.py

```diff
@@ -0,0 +1,41 @@
+def s(l,st,end):
+  if len(l) <= 1: return
+  m=len(l)//2
+  ll=l[:m]
+  rr=l[m:]
+  s(ll,st,m)
+  s(rr,m+1,end)
+  i=0
+  j=0
+  k=0
+  while i < len(ll) and j< len(rr):
+    print('i', i)
+    print('j', j)
+    print('len(ll)', len(ll))
+    print('len(rr)', len(rr))
+    print('ll[i]', ll[i])
+    print ('rr[j]', rr[j])
+
+    if ll[i]>rr[j]:
+      l[k] = rr[j]
+      j+=1
+    if ll[i]<=rr[j]:
```

Comment:
this condition is redundant. 
also if the first condition passes, j will be modified and you will be checking the wrong element or have a buffer overflow.